src/middlewared/middlewared/plugins/fc/fc_host.py

```diff
@@ -0,0 +1,378 @@
+
+    async def _get_remote_fc_hosts(self):
+        physical_port_filter = [["physical", "=", True]]
+        try:
+            return await self.middleware.call('failover.call_remote', 'fc.fc_hosts', [physical_port_filter])
+        except CallError as e:
+            if e.errno in NETWORK_ERRORS + (CallError.ENOMETHOD,):
+                # swallow error, but don't present any choices
+                return []
+            else:
+                raise
+
+    @private
+    async def wire(self, overwrite=False):
+        """
+        Wire up fc_host pairings, using the 'slot' field from fc.fc_hosts
+        to determine which pairs go together.
+
+        Need to handle the case where the other controller is currently
+        down and is therefore unable to give its current config.
+        """
+        if not await self.middleware.call('fc.capable'):
+            return False
+        if await self.middleware.call('failover.licensed'):
+            # HA
+            if (fo_status := await self.middleware.call('failover.status')) != 'MASTER':
+                self.logger.info('Cannot wire fc_host if not ACTIVE node: %r', fo_status)
+                # If we're the BACKUP node then we don't need to keep trying
+                if fo_status == 'BACKUP':
+                    self.do_reset_wired = True
+                    return True
+                return False
+            node = await self.middleware.call('failover.node')
+            slot_2_fc_host_wwpn = defaultdict(dict)
+            physical_port_filter = [["physical", "=", True]]
+            match node:
+                case 'A':
+                    node_a_fc_hosts = await self.middleware.call('fc.fc_hosts', physical_port_filter)
+                    node_b_fc_hosts = await self._get_remote_fc_hosts()
+                case 'B':
+                    node_a_fc_hosts = await self._get_remote_fc_hosts()
+                    node_b_fc_hosts = await self.middleware.call('fc.fc_hosts', physical_port_filter)
+                case _:
+                    raise CallError('Cannot configure FC until HA is configured')
+            # Match pairs by slow (includes PCI function)
+            for fc_host in node_a_fc_hosts:
+                slot_2_fc_host_wwpn[fc_host['slot']]['A'] = str_to_naa(fc_host.get('port_name'))
+            for fc_host in node_b_fc_hosts:
+                slot_2_fc_host_wwpn[fc_host['slot']]['B'] = str_to_naa(fc_host.get('port_name'))
+            # Iterate over each slot and make sure the corresponding fc_host entry is preent
+            sorted_slots = sorted(list(slot_2_fc_host_wwpn.keys()))
+            result = True
+            for slot in sorted_slots:
+                wwpn = slot_2_fc_host_wwpn[slot].get('A')
+                wwpn_b = slot_2_fc_host_wwpn[slot].get('B')
+                if wwpn and wwpn_b:
+                    existing = await self.middleware.call('fc.fc_host.query', [
+                        ['OR', [['wwpn', '=', wwpn], ['wwpn_b', '=', wwpn_b]]]
```

Comment:
Rather than `OR` are we able to do `in` ?

Reply:
```
[['wwpn', 'in', [wwpn, wwpn_b]]]
```

Reply:
I don't think so.  The keys are also different, not just the values.